src/main/java/seedu/address/logic/parser/eventcommandparsers/EventEditCommandParser.java

```diff
@@ -0,0 +1,136 @@
+            editEventDescriptor.setBudget(ParserUtil.parseBudget(argMultimap.getValue(PREFIX_BUDGET).get()));
+        }
+
+        if (argMultimap.getValue(PREFIX_START_DATETIME).isPresent()) {
+            startDate = ParserUtil.parseDateAndTime(argMultimap.getValue(PREFIX_START_DATETIME).get());
+            editEventDescriptor.setStartDate(startDate);
+        } else {
+            startDate = null;
+        }
+        if (argMultimap.getValue(PREFIX_END_DATETIME).isPresent()) {
+            endDate = ParserUtil.parseDateAndTime(argMultimap.getValue(PREFIX_END_DATETIME).get());
+            editEventDescriptor.setEndDate(endDate);
+        } else {
+            endDate = null;
+        }
+
+        //compare end datetime from user to ensure it is after/same as start datetime from user
+        if (startDate != null && endDate != null && endDate.dateAndTime.isBefore(startDate.dateAndTime)) {
+            throw new ParseException(MESSAGE_INVALID_DATE_PARAMS);
+        }
+
+        parseRolesForEdit(argMultimap.getAllValues(PREFIX_ROLE)).ifPresent(editEventDescriptor::setRoles);
+        parseMaterialsForEdit(argMultimap.getAllValues(PREFIX_MATERIAL)).ifPresent(editEventDescriptor::setMaterials);
+
+        if (!editEventDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EventEditCommand.MESSAGE_NOT_EDITED);
+        }
+
+        return new EventEditCommand(index, editEventDescriptor);
+    }
+
+    /**
+     * Parses {@code Collection<String> roles} into a {@code Set<Role>} if {@code roles} is non-empty.
+     * If {@code roles} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Role>} containing zero skills.
+     */
+    private Optional<Set<Role>> parseRolesForEdit(Collection<String> roles) throws ParseException {
+        assert roles != null;
+
+        if (roles.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> roleSet = roles.size() == 1 && roles.contains("") ? Collections.emptySet() : roles;
+        return Optional.of(ParserUtil.parseRoles(roleSet));
+    }
+
+    /**
+     * Parses {@code Collection<String> materials} into a {@code Set<Material>} if {@code materials} is non-empty.
+     * If {@code materials} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Material>} containing zero materials.
+     */
+    private Optional<Set<Material>> parseMaterialsForEdit(Collection<String> materials) throws ParseException {
+        assert materials != null;
+
+        if (materials.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> materialSet = materials.size() == 1 && materials.contains("") ? Collections.emptySet()
+                                                                                            : materials;
+        return Optional.of(ParserUtil.parseMaterials(materialSet));
+    }
```

Comment:
not sure but for your reference if some test cases fail due to the change in Material

Reply:
unless you wanna make more changes

Reply:
nah it is not ur fault, i lost my cool.